mkosi/run.py

```diff
@@ -514,36 +519,49 @@ def sandbox_cmd(
 
         if d and not any(Path(d).is_relative_to(dir) for dir in (*dirs, "/usr", "/nix", "/tmp")):
             cmdline += ["--bind", d, d]
+    else:
+        cmdline += ["--dir", "/var/tmp", "--unshare-ipc"]
 
-    for d in ("bin", "sbin", "lib", "lib32", "lib64"):
-        if (p := tools / d).is_symlink():
-            cmdline += ["--symlink", p.readlink(), Path("/") / p.relative_to(tools)]
-        elif p.is_dir():
-            cmdline += ["--ro-bind", p, Path("/") / p.relative_to(tools)]
+        if devices:
+            cmdline += ["--bind", "/sys", "/sys", "--bind", "/dev", "/dev"]
+        else:
+            cmdline += ["--dev", "/dev"]
 
-    path = "/usr/bin:/usr/sbin" if tools != Path("/") else os.environ["PATH"]
+        if network and Path("/etc/resolv.conf").exists():
+            cmdline += ["--ro-bind", "/etc/resolv.conf", "/etc/resolv.conf"]
 
-    cmdline += ["--setenv", "PATH", f"/scripts:{path}", *options]
+    path = "/usr/bin:/usr/sbin" if tools != Path("/") else os.environ["PATH"]
 
-    # If we're using /usr from a tools tree, we have to use /etc/alternatives from the tools tree as well if it
-    # exists since that points directly back to /usr. Apply this after the options so the caller can mount
-    # something else to /etc without overriding this mount. In relaxed mode, we only do this if /etc/alternatives
-    # already exists on the host as otherwise we'd modify the host's /etc by creating the mountpoint ourselves (or
-    # fail when trying to create it).
-    if (tools / "etc/alternatives").exists() and (not relaxed or Path("/etc/alternatives").exists()):
-        cmdline += ["--ro-bind", tools / "etc/alternatives", "/etc/alternatives"]
+    cmdline += ["--setenv", "PATH", f"/scripts:{path}"]
 
     if scripts:
         cmdline += ["--ro-bind", scripts, "/scripts"]
 
-    if network and not relaxed and Path("/etc/resolv.conf").exists():
-        cmdline += ["--ro-bind", "/etc/resolv.conf", "/etc/resolv.conf"]
+    with contextlib.ExitStack() as stack:
+        for d in ("etc", "opt", "srv", "media", "mnt", "var", "run", "tmp"):
+            if not overlay and not relaxed:
+                continue
```

Comment:
this short-circuits the whole loop body, so maybe this could be hoisted up to yield immediately in that case?